lib/backend.js

```diff
@@ -441,20 +441,33 @@ exports.createServer = (dynamodb, docClient) => {
 
         // Create a variable to uniquely identify each expression attribute
         let i = 0
+
         for (const key in filters) {
           if (filters[key].type === 'N') {
             filters[key].value = Number(filters[key].value)
           }
+
           ExpressionAttributeNames[`#key${i}`] = key
           ExpressionAttributeValues[`:key${i}`] = filters[key].value
-          if (indexBeingUsed && indexBeingUsed.KeySchema.find(
-            (keySchemaItem) => keySchemaItem.AttributeName === key)
-          ) {
-            KeyConditionExpression.push(`#key${i} ${filters[key].operator} :key${i}`)
+          const matchedKeySchema = indexBeingUsed !== undefined ? indexBeingUsed.KeySchema.find(
```

Comment:
`indexBeingUsed` starts as `null` so this needs to be:
```suggestion
          const matchedKeySchema = indexBeingUsed !== null ? indexBeingUsed.KeySchema.find(
```
or just a truthy check like before.